addons/mail/static/tests/messaging_menu/messaging_menu.test.js

```diff
@@ -1315,3 +1315,25 @@ test("failure is removed from messaging menu when message is deleted", async ()
     pyEnv["mail.message"].unlink([messageId]);
     await contains(".o-mail-NotificationItem", { count: 0 });
 });
+
+test("user notification from inbox redirect to discuss inbox", async () => {
+    const pyEnv = await startServer();
+    const messageId = pyEnv["mail.message"].create({
+        body: "Hello world!",
+        message_type: "user_notification",
+        model: "res.partner",
+        needaction: true,
+        res_id: serverState.partnerId,
+    });
+    pyEnv["mail.notification"].create({
+        mail_message_id: messageId,
+        notification_status: "sent",
+        notification_type: "inbox",
+        res_partner_id: serverState.partnerId,
+    });
+    await start();
+    await click(".o_menu_systray i[aria-label='Messages']");
+    await click(".o-mail-NotificationItem:contains('Hello world!')");
+    await contains(".o-mail-Discuss-threadName[title='Inbox']");
+    await contains(".o-mail-Message.o-highlighted", { text: "Hello world!" });
```

Comment:
Note: `:text()` selector has been recently merged in HOOT so we no longer need to use a separate parameter.